### ISSUE TITLE
Discuss color contrast rounding to align with other tools

### ISSUE THREAD

**Issue by marcysutton** [dequelabs/axe-core#494]
When color contrast ratios fall between 4.4 and 4.5, values that are extremely close to 4.5:1 get rounded down, and therefore fail the color-contrast check. We discussed this previously when a bug was uncovered in axe-core where it would return *violations* with a ratio of 4.5:1. The color-contrast check was rounding up for displaying results, but basing its analysis on the raw number (something like 4.4889989899). We decided [at that time](https://github.com/dequelabs/axe-core/commit/062ffabaa42952a75f14ae48552756b5ca09efd6) to round down.

```
// lib/checks/color/color-contrast.js
var cr = axe.commons.color.hasValidContrastRatio(bgColor, fgColor, fontSize, bold);

// truncate ratio to three digits while rounding down
// 4.499 = 4.49, 4.019 = 4.01
var truncatedResult = Math.floor(cr.contrastRatio * 100) / 100;
```

This came up again this week, where a particular color combination that passes all other contrast tools is failing in axe-core since it falls just shy of 4.5:1:

`Element has insufficient color contrast of 4.49 (foreground color: #ffffff, background color: #0078d7, font size: 10.5pt, font weight: normal)`

The full contrast ratio for those colors and font-size is 4.498861479739532. I think we should pass this, as it's so close you wouldn't be able to tell the difference in a passing ratio by looking at it. If we don't want to pass colors closer to 4.4, we could potentially split the decimal point and round up only if it's above 4.45:1, or something. It's pretty arbitrary at that point. 

What do you think @WilcoFiers? You mentioned when we discussed rounding originally that it had been discussed on a W3C call but nobody had a huge preference. Was there a conclusion after that?

**Comment by liloda** [comment]
I think rounding up when it's this close is a good idea.  It's a losing proposition to convince design teams to change their color palette when other testing tools, especially the venerable Colour Contrast Analyser, pass a particular color combination.

**Comment by nschonni** [comment]
Here's the ARIA discussion on rounding https://github.com/w3c/wcag/issues/200

**Comment by WilcoFiers** [comment]
Not sure why this got reopened. It doesn't really matter what other tools do. The AG Working group decided that `(eg. 4.499:1 would not meet the 4.5:1 threshold)`, and so that's what we should do too.

https://github.com/w3c/wcag/pull/248/files?diff=split#diff-43287e4aaf75e325fe20279db4425509R17

**Comment by marcysutton** [comment]
Thanks Wilco, that answers it. This issue was reopened based on a question from Microsoft, but that's a pretty clear guideline. We don't have to change anything as a result (the other tools do!).
